File: sort/bubble_sort/c++/BubbleSort.cpp
#include<iostream>
using namespace std;

void bubble_sort( int A[ ], int n ) {
    int temp;
    for(int k = 0; k< n-1; k++) {
        // (n-k-1) is for ignoring comparisons of elements which have already been compared in earlier iterations

        for(int i = 0; i < n-k-1; i++) {
            if(A[ i ] > A[ i+1] ) {
                // here swapping of positions is being done.
                temp = A[ i ];
                A[ i ] = A[ i+1 ];
                A[ i + 1] = temp;
            }
        }
    }
}

int main()
{
    int n;
    cout<<"Enter size of array: ";
    cin>>n;
    int arr[n];
    cout<<"Enter elements in array: ";
    for(int i=0;i<n;i++)
    {
        cin>>arr[i];
    }
    bubble_sort(arr,n);
    cout<<"Sorted array: "<<endl;
    for(int i=0;i<n;i++)
    {
        cout<<arr[i]<<endl;;
    }
}

File: sort/bubble_sort/php/bubble_sort.php
<?php

function bubbleSort(array $elements)
{
    $elementCount = count($elements);

    for ($i = 1; $i < $elementCount; $i++) {
        for ($j = 0; $j < $elementCount - $i; $j++) {
            if ($elements[$j] > $elements[$j + 1]) {
                $temp = $elements[$j];
                $elements[$j] = $elements[$j + 1];
                $elements[$j + 1] = $temp;
            }
        }
    }

    return $elements;
}
